cmd/controller-manager/app/controller_manager.go

```diff
@@ -111,21 +112,30 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		rest.AddUserAgent(k8sKubeconfig, controllerManagerAgentName),
 	)
 	if err != nil {
-		glog.Fatalf("Invalid k8s API configuration: %v", err)
+		return fmt.Errorf("invalid Kubernetes API configuration: %v", err)
 	}
 
 	glog.V(4).Infof("Building service-catalog kubeconfig for url: %v\n", controllerManagerOptions.ServiceCatalogAPIServerURL)
+
+	var serviceCatalogKubeconfig *rest.Config
 	// Build the service-catalog kubeconfig / clientBuilder
-	serviceCatalogKubeconfig, err := clientcmd.BuildConfigFromFlags(
-		controllerManagerOptions.ServiceCatalogAPIServerURL,
-		// controllerManagerOptions.ServiceCatalogKubeconfigPath,
-		"", // TODO: tolerate missing kubeconfig
-	)
+	if controllerManagerOptions.ServiceCatalogAPIServerURL == "" && controllerManagerOptions.ServiceCatalogKubeconfigPath == "" {
+		// explicitly fall back to InClusterConfig, assuming we're talking to an API server which does aggregation
```

Comment:
If we did that, we'd get
```
Using inClusterConfig to talk to service catalog API server -- make sure your API server is registered with the aggregator
Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
```

Not only is the second message a bit redundant, it's also confusing and incorrect (it mentions flags that don't exist).  I figured we'd rather avoid that.

The fallback path is fairly well-defined in BuildConfigFromFlags, it's just not a good message.